[docs] Forms Documentation Improvement

dashtinejad opened the pull request:
I feel that the forms documentation is missing some important and fundamentals inputs:

- multi select
- checkbox
- radio buttons

So, I've added them to the documentation to clarify how to convert those form inputs to React Controlled Components.

joh1234 commented:
Update README
https://goo.gl/JBljdR

or 
http://bit.ly/2dK6Ojk

lacker commented on docs/docs/forms.md:
````diff
@@ -174,6 +174,132 @@ class FlavorForm extends React.Component {
+<input type="checkbox" />
+```
+
+You can change this checkbox to controlled component by setting the `checked` property,
````
I think it would be better to combine the JSX and javascript code in this example. Look at how textarea works, the format is just:

--
In HTML a textarea is like this:

(example)

In React, you change X and Y, so a full example is this:

(full example)

Note some brief stuff stuff.
-- 

How about using that parallel format for checkbox and radio buttons?

lacker commented:
Originally I was thinking, meh we don't need these examples, but by now a number of folks have suggested re-adding them so I officially change my mind. So it would be cool to add these examples, I just suggest using a format equivalent to the way "textarea" works where the example is a full one rather than describing it bit by bit - I added comments to that effect inline

dashtinejad commented:
@lacker Good points, I'll work on it.

lacker commented:
AFAICT this is still in progress rather than needing review so I tagged it back. Let me know if that is not the case though

dashtinejad commented:
@lacker As you suggested, I've added 2 new sections (for checkbox and radio) using a format equivalent to the way "textarea" where the example is a full one. Also I've added a note about `multi` attribute for `select` tag. Please let me know if I missed something.

dashtinejad commented:
@lacker @gaearon @sebmarkbage I really do believe that handling `checkbox` and `radio` elements should be on official documentation. Please let me know If I should improve my suggested documentation.

bvaughn commented:
Thank you for submitting this PR! 😄 

Unfortunately the documentation and source code for reactjs.org has been moved to a different repository: https://github.com/reactjs/reactjs.org

I realize this PR is old and there may no longer be interest in seeing it merged. If there is, please open a new PR there. Sorry for the inconvenience!

(For more backstory on why we moved to a new repo, see issue #11075)
